### DISCUSSION TITLE
Final decision: "Project Reunion" requires Windows 10 1809/17763/RS5 or later

### DISCUSSION THREAD

**Discussion by bitcrazed** [microsoft/WindowsAppSDK#413] [Final decisions]
_We'd love your feedback on this topic, we'll be making a decision midday PST on 2/9, please let us know if you have any feedback before then!_

## Final decision (2/10/2021): On Windows 10, Reunion will only support down to RS5

We appreciate all feedback that has been shared! After careful consideration of feedback here, feedback from customer meetings, and internal data, we feel the best decision is to only support down to RS5 on Windows 10. This will certainly cause some pain to customers, however when balancing all options, this seems to be the best option for balancing engineering constraints with what most customers need.

However, for Windows 7/8, we will be leaving that topic open to be discussed and decided at a later date.

## Rationale

We aim for "Project Reunion" to support as broad a user-base as possible whilst focusing as many of our engineering resources as possible toward fulfilling all goals and objectives.

Having carefully considered telemetry and customer's needs that we've heard so far, balanced against goals, plans, product lifecycles, engineering requirements, etc., we propose that "Project Reunion" will require/support developers and users running Windows 10 RS5/1809/17763 or later.

> If this policy causes you particular difficulties or challenges, please do share in the replies below. We appreciate and will consider all constructive feedback.

### 1. Why RS5 and not RS4, or earlier releases of Windows 10.

RS5 included substantial improvements throughout the OS that reduced latency, improved performance and responsiveness, security, reliability, etc. If we were to start at RS4, we would have to compromise on all of these and other qualities of the platform, plus we'd have to spend significant engineering effort porting many of the required changes back to RS4, rather than spending that effort expanding and improving "Project Reunion" itself. This would be self-defeating.

Also, whilst not our primary consideration, it is important to note that RS4 is no longer receiving mainstream nor extended support. Thus, setting an expectation that "Project Reunion" would support an unsupported OS release would be difficult.

In addition, RS5 is the base of our current Long Term Servicing Channel (LTSC) release for those customers who require a highly stable OS release offering considerably extended support policies.

### 2. Will "Project Reunion" support Windows 7 or 8?

No: "Project Reunion" requires many OS/platform features that did not exist/ship within Windows 7, and many that changed substantially. Porting all of the required features and changes back to Windows 7/8 would result in us shipping most of Windows 10 again. And could break existing apps etc., which we'd rather avoid.

**Comment by riverar** [comment]
I think this makes sense, but now there's a disconnect between Project Reunion and WinUI 3. WinUI 3 is currently targeting Windows 10 April 2018 Update (version 1803) as its minimum. It sounds like perhaps after _midday PST on 2/9_ we're going to hear WinUI 3 plans have changed.

**Reply by saint4eva** [reply]
> _We'd love your feedback on this topic, we'll be making a decision midday PST on 2/9, please let us know if you have any feedback before then!_
> 

I would suggest you start from 1903 or later.

**Reply by bitcrazed** [reply]
Thanks for the feedback Rafael. We hear you and are working hard on rationalizing and clarifying the versioning and support policies for "Project Reunion" and its various component parts. We aim to share some more clarity in new proposals soon. Stay tuned.

**Comment by simonferquel** [comment]
Hmm, looking at our user-base on Docker Desktop, and it seems that there are still a significant 1803 users out there (around 3% of our total users - and it is not really declining). Depending on when Reunion is ready and how numbers evolve until then, that could be a blocker.

**Reply by simonferquel** [reply]
3 % being 1803 only. If we count versions prior to 1803 it is 5%. That might not look high, but in absolute numbers it is still significant.

**Reply by bitcrazed** [reply]
Thanks for the info Simon. Agree that % often mask absolute numbers. However, we expect that users will continue to roll forward onto newer OS releases over time, but there will always be a declining number of users who cannot / will not upgrade for a variety of reasons, even if that means they end up running on unsupported OS releases. There are, after all, still users running Windows 2000 and XP in production in 2021!

While we respect their choice to do so, we want to avoid impeding the benefits of decoupling the Windows app platform from the OS for the vast majority of developers & users who run supported Windows releases. This will become increasingly apparent as we continue to release "Project Reunion" updates multiple times a year, lifting and adding more and more of the platform into OS-independent* packages.

We hope that Reunion will deliver sufficient benefits to app developers and their users that upgrading to at least supported versions of Windows will be compelling.

(* OS independent: For OS releases at or newer than the "Project Reunion" base supported OS version)

**Comment by asklar** [comment]
From a React Native Windows standpoint, I'd like to ensure we can eventually move to Reunion/WinUI 3 and not lose out on supported versions. @riverar mentioned a disconnect with WinUI 3, which I will echo here too. 
It's important that our offerings all up are aligned as much as possible and we don't continue to fragment the ecosystem with a la carte options that are slightly/subtly incompatible with each other. In other words: there should be a golden path across the portfolio of technologies we offer, at least at the dxp level.

**Reply by bitcrazed** [reply]
Appreciate your and @riverar's point - we're working hard to rationalize and clarify our base support policy (hence this discussion post) as well as how and when we version various component parts of "Project Reunion". 

We'll be sharing many more proposals discussing these and other interrelated plans, policies, etc. in the coming months. Keep the feedback coming :)

**Comment by Tourniquet88** [comment]
To be honest I would have thought you would at least go back to RS1 / 14393 / Anniversary Update as this is a) LTSC release, b) the last release for most of the Windows 8 Tablets (Clovertrail support). What's more is, if you support 14393 you'd also cover basically all device types there are.

Also we have seen how the performance of Windows 10 was degraded from Version to Version so it would be nice to have "good" Version supported. 14393 will still get updates until 13.10.2026 which is quite a long time. And then there is also a Server and IoT Enterprise version of it that has the same support life-cycle.

However if it's really only between 17763 and 17134 than it doesn't really matter. As it was mentioned already. 17134 support end is near and with 17763 being LTSC it does make more sense.

**Reply by MarkIngramUK** [reply]
1809 seems like a good starting point. You've got to pick a cut off point, so naturally some people will always be disappointed, but I would hate to see Project Reunion compromised in quality or performance (or to take time away from features as time must be spent on porting features and performance back to earlier versions).

**Reply by bitcrazed** [reply]
@Tourniquet88 Thanks for sharing your thoughts.

There are a mountain of features and improvements to Windows 10 between RS1 and RS5. Porting those missing features all the way back to RS1 would take years and would prevent us from delivering "Project Reunion" at all until after such a port was complete - likely long after even the LTSC support ends!

As far as we are aware, none of the Windows 8 era laptops, tablets, etc. are supported by their OEM, let alone by Windows - mainstream support for RS1 ended in April 2018 - 3 years ago. 

Also note that LTSC releases intentionally receive few new features, typically other than important management/security improvements.

**Reply by Tourniquet88** [reply]
@bitcrazed https://www.zdnet.com/article/microsoft-agrees-to-extend-support-deadline-for-clover-trail-pcs/ (https://answers.microsoft.com/{lang-locale}/windows/forum/windows_10-windows_install/intel-clover-trail-processors-are-not-supported-on/ed1823d3-c82c-4d7f-ba9d-43ecbcf526e9) 

"To keep our customers secure, we will provide security updates to **these specific devices running the Windows 10 Anniversary Update until January of 2023**, which aligns with the original Windows 8.1 extended support period."

But yeah I get that RS1 would be a massive undertaking.

**Comment by Marv51** [comment]
By 5/11/2021 are security updates available to some organizations on versions of Windows before 1909? 

And related, would an LTSC version of Windows run a WinUI3/Reunion app?

**Reply by bitcrazed** [reply]
As per [Microsoft's product lifecycle FAQ](https://docs.microsoft.com/en-us/lifecycle/faq/windows#when-a-windows-operating-system-os-reaches-the-end-of-its-lifecycle-or-is-no-longer-supported-does-that-mean-new-programs-will-not-run-on-that-os), "Once a Microsoft operating system (OS) reaches the end of support, customers will no longer receive security updates".

By May 2021, the only releases of Windows 10 prior to 1909 receiving security updates will be those customers who run supported LTSC releases.

There's a good summary of various releases and support lifecycles here: https://en.wikipedia.org/wiki/Windows_10_version_history#Channels

**Reply by prayaas-a** [reply]
Well it looks like 1809 (which is the latest LTSC) would run WinUI3 as well as Reunion apps if the current proposal is greenlit.

**Comment by wjk** [comment]
My 2¢: I use 18362 as the minimum baseline for my projects (although there is little to no reasoning behind this choice; it’s just the build number I remember). However, I believe it is of paramount importance that Reunion support _every_ currently-supported variant of Windows, including all server and LTSC editions.

Once a Windows 10 release loses all support (security updates, everything), then _newer_ versions of Reunion need not be compatible it. Older versions of the components must continue to work on the OS versions and variants they were originally designed to run on, so that I can provide a separate version of my app that uses older components than the mainline branch if I need to continue down-targeting to those versions.

**Reply by prayaas-a** [reply]
Generally I would agree with this and it would help them limit Reunion to 1809 which is supported till this May. But is Project Reunion even going to be RTM by May 2021? If we go by supported OS versions, assuming that Project Reunion RTMs by early 2022, only 1909 and above should be supported (1909 would still be in Enterprise/Education support but already out of date for consumers).
It's also important not to slow down development of the project so much just to support older Windows versions which anyway become unsupported by the time development finishes.

**Comment by Marv51** [comment]
Can someone clarify the current thinking about how this requirement is expected to evolve in the future?

Are you expecting to increase this minimum in the near future? What would the process be like and what are the considerations for such an increase?
Is Windows usage the deciding factor here, too?


(Personally, our product does support LTSC versions and versions that are not supported by Microsoft)

**Reply by bitcrazed** [reply]
@Marv51 Great questions. We're still figuring out the versioning and evolution story in more detail and will share for review once we have something stable.

However, the idea here is to set a starting point for the platform, APIs, and capabilities that "Project Reunion", and thus apps that depend upon "Project Reunion". In the future, we aim to support as broad a set of features and capabilities on as broad a set of OS releases as possible. 

Many future features and APIs etc. will be supported all the way back to 1809 or later. However, there may well be some future features that are only supported on later OS releases. In this case, their minimum OS release requirements will be clearly articulated via the API package's metadata and honored by the build/IDE tooling, etc. 

Importantly, however, we aim to ensure that existing API packages will run on all future OS releases, insulating your apps from underlying OS releases and changes, increasing your apps' longevity and support.

A sound package/API versioning strategy will help reduce compatibility guesswork, and enable clearer, more consistent, more predictable versioning support. But that'll be the subject of future proposals, where we look forward to hearing more of your thoughts & suggestions. Stay tuned!

**Reply by riverar** [reply]
@bitcrazed Your comment: "However, there may well be some future features that are only supported on later OS releases." scares me. It sounds like we're going down the wrong path again. This is one of the problems that Reunion was supposed to fix--the introduction of newer APIs that we can't use downlevel. It was my understanding that polyfills would fill in gaps where needed to realize this vision.

**Reply by bitcrazed** [reply]
@riverar Wherever we possibly can we'll polyfill new capabilities on all supported OS versions. 

But thinking really long term, might there be scenarios in which there's a fundamental shift in technology (perhaps a hardware dependency?) that simply cannot be polyfilled/emulated/etc. on older OS versions? Perhaps, so I don't think we can rule this out.

**Comment by andrewleader** [comment]
## Final decision (2/10/2021): On Windows 10, "Project Reunion" will support Windows 10 1809/17763/RS5 or later

We appreciate all feedback that has been shared! After careful consideration of feedback here, feedback from customer meetings, and internal data, we feel the best decision is to only support down to RS5 on Windows 10. This will certainly cause some pain to customers, however when balancing all options, this seems to be the best option for balancing engineering constraints with what most customers need.

However, for Windows 7/8, we will be leaving that topic open to be discussed and decided at a later date (we will post a discussion when we start to consider that topic).

**Reply by MarkIngramUK** [reply]
Onwards and upwards!
